modules/ROOT/pages/propagate-metadata-studio.adoc

```diff
@@ -0,0 +1,161 @@
++
+image::extract-operation-metadata.png[]
+. Name your target flow `flow-2`.
+. Select *Metadata Preferences* to configure the input and output metadata. +
+Note that Studio automatically recognizes the metadata of your flow.
++
+image::extract-flow-metadata.png[]
+. Select *OK*.
+. Your Logger component is now in `flow-2`, the metadata tab is populated with autogenerated metadata, and the input tab in the metadata explorer reflects the payload format that the Set Payload component configures in `flow-1`.
++
+image::extract-flow-metadata-final.png[]
+
+
+== Propagate Metadata to a Flow With Custom Metadata Already Configured
+
+When you add a Flow Reference component and reference another flow that has custom metadata already configured, Studio prompts you to select the metadata that you want to propagate.
+
+. In the taskbar at the top of the Anypoint Studio display, select *File* > *New* > *Mule Project*.
+. Type a name for this project, and select *Finish*.
+. Drag two flow components to your empty canvas and name them `flow-1` and `flow-2` respectively.
++
+image::both-flows-metadata.png[]
+. In `flow-1`, drag a Set Payload component and configure the payload as:
++
+[source,JSON,linenums]
+--
+{
+  "value": {
+    "ID": 1,
+    "departureDate": "2016/12/20",
+    "emptySeats": 200,
+    "plane": {
+      "type": "Boeing 747",
+      "totalSeats": 400
+    }
+  }
+}
+--
++
+image::set-payload-config.png[]
+. Create a JSON file with the name _arrival-type-example.json_ in your `src/main/resources/examples` directory with the following format:
++
+.arrival-type-example.json
+[source,JSON,linenums]
+--
+{
+  "value": {
+    "ID": 1,
+    "arrivalDate": "2016/12/20",
+    "plane": {
+      "type": "Boeing 747",
+      "totalSeats": 400
+    }
+  }
+}
+--
+. Select `flow-2` in your canvas, select the *Metadata* tab, and *Add Metadata*:
++
+image::add-metadata-flow-2.png[]
+. Select *Input: Payload*, and then select the *Edit*.
++
+image::configure-input-metadata.png[]
+.  Select *Add*, create a new metadata type called _arrivalType_, and select *Create type*.
++
+image::create-metadata-type.png[]
+. Select the type *JSON* > *Example* and select the *...* button to browse to `src/main/resources/examples/arrival-type-example.json`
++
+image::create-json-metadata-type.png[]
+. Click *Select*.
+. Drag a Logger component to `flow-2`, and configure it to log the `payload`.
++
+image::logger-payload-metadata.png[]
+. Drag a Flow Reference component to the end of `flow-1`, and configure it to reference _flow-2_:
++
+image::flow-reference-metadata.png[]
+. Select the Flow Reference component *Metadata* tab, and select *Set Metadata*.
++
+image::set-metadata-example.png[]
++
+Studio warns you that setting the metadata from `flow-1` will overwrite the existing `arrivalType` metadata type configured for `flow-2`.
++
+image::overwrite-metadata.png[]
+. Select *Customize* to configure the desired metadata to propagate to the flow:
```

Comment:
Colon only if the end of its sentence leads directly to the image, without any text between.
```suggestion
. Select *Customize* to configure the desired metadata to propagate to the flow.
```

Reply:
(You can avoid this by moving the second sentence to follow the image. See also Lines 76-79.)